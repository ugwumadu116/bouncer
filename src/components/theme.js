import * as colors from "./color/color";
import * as fontsize from "./text/fontsize";
import * as fontweight from "./text/fontweight";
import * as lineheight from "./text/lineheight";
import * as letterspacing from "./text/letterspacing";
import * as btnSizes from "./button/button";
import * as btnState from "./button/btnState";
import * as inputsizes from './form/input/inputsizes'
import * as inputbuttonSize from './form/inputButton/inputbuttonsizes'
import * as selectSize from './form/select/selectSizes'


const theme = {
  colors,
  fontsize,
  fontweight,
  lineheight,
  letterspacing,
  btnSizes,
  btnState,
  inputsizes,
  inputbuttonSize,
  selectSize
};
export default theme;
