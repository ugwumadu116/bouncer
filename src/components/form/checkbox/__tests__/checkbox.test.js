import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'
import CheckGroup from '../index'
import theme from '../../../theme'
import { render } from '@testing-library/react'

import '@testing-library/jest-dom/extend-expect'



it('renders correctly', ()=>{
    const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <CheckGroup/>  
    </ThemeProvider>
    
    
    ).toJSON();
    expect(tree).toMatchSnapshot();
})


describe('Checkbox', () => {
  it('find label', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <CheckGroup checked={true} handleClick={()=>{}}/>
      </ThemeProvider>
    
    );
    const elem = getByTestId('box');
    expect(elem).toHaveProperty('checked', true)
  })
})