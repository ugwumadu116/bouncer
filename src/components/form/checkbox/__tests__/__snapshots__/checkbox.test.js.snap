// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="sc-htpNat hCtDsA"
>
  <label
    className="sc-bxivhb fVcWOf"
    htmlFor="checkbox"
  />
  <div
    className="sc-bdVaJa bBPEUt"
    id="checkbox"
  >
    <input
      aria-label="checkbox"
      aria-required="true"
      className="sc-ifAKCX MhoXa"
      data-testid="box"
      type="checkbox"
    />
    <div
      className="sc-EHOje iMWpYg"
    >
      <svg
        className="sc-bwzfXH bkYyAD"
        viewBox="0 0 24 24"
      >
        <polyline
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
  </div>
</span>
`;
